radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/ConstraintMachine.java

```diff
@@ -387,7 +389,14 @@ public Particle next() {
 
 					expectEnd = false;
 				} else if (inst.getMicroOp() == REInstruction.REMicroOp.SIG) {
-					metering.onSigInstruction(context);
+					if (context.permissionLevel() != PermissionLevel.SYSTEM) {
+						metering.onSigInstruction(context);
+					}
+				} else {
+					// Collect no-ops here
+					if (inst.getMicroOp() != REInstruction.REMicroOp.MSG) {
+						throw new ProcedureException("Unknown op " + inst.getMicroOp());
+					}
```

Comment:
We are missing `HEADER`

Reply:
oof, good catch, need to add some tests with `HEADER`